Terminal.Gui/Application.cs

```diff
@@ -494,7 +494,7 @@ public static RunState Begin (Toplevel Toplevel)
 				OverlappedTop?.OnChildLoaded (Toplevel);
 				Toplevel.OnLoaded ();
 				Toplevel.SetNeedsDisplay ();
-				Toplevel.Redraw (Toplevel.Bounds);
+				Toplevel.Draw ();
```

Comment:
@tig the only inconvenient I see here is when a view use the `OnDrawContentComplete` to draw after all drawing stuff is done. If you call `OnDrawLineCanvas` on the `DeferredDraw` will screw-up the last draw by the view, unless we may use the `OnDeferredDrawComplete` instead, right?

Reply:
> @tig the only inconvenient I see here is when a view use the `OnDrawContentComplete` to draw after all drawing stuff is done. If you call `OnDrawLineCanvas` on the `DeferredDraw` will screw-up the last draw by the view, unless we may use the `OnDeferredDrawComplete` instead, right?

I think this is the right model:

- `Draw/OnDraw/OnDrawComplete` only involve `Bounds`. They have nothing to do with the `Frames` that are outside of the `Bounds`. In these methods it is not allowed/possilbe/ok (not enforced) to draw anywhere but within `Bounds`.
- `DrawDeferred/OnDrawDeferred/OnDrawDefferedComplete` and `LineCanvas` only involve the `Thickness` defined by `Margin`, `Border`, and `Padding`.  These methods will not (should not) draw anywhere but in that `Thickness` (and thus NOT in `View.Bounds`).

Does this help?

Reply:
@tig but why complicating this even more by adding one more method when we can use all in the `Draw` method with the right sequence? The main reason for I stated this is the need of call one additional method and probably will cause a tons of unit tests failing because the `DeferredDraw` method wasn't called somewhere.